compiler/rustc_attr_parsing/src/attributes/stability.rs

```diff
@@ -24,11 +24,16 @@ pub fn find_stability(
     item_sp: Span,
 ) -> Option<(Stability, Span)> {
     let mut stab: Option<(Stability, Span)> = None;
-    let mut allowed_through_unstable_modules = false;
+    let mut allowed_through_unstable_modules = None;
 
     for attr in attrs {
         match attr.name_or_empty() {
-            sym::rustc_allowed_through_unstable_modules => allowed_through_unstable_modules = true,
+            sym::rustc_allowed_through_unstable_modules => {
+                allowed_through_unstable_modules = Some(match attr.value_str() {
```

Comment:
(to implement the way I suggested before)

you should use [`meta_item_list`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_ast/attr/trait.AttributeExt.html#tymethod.meta_item_list), find one that has name `deprecated`, then use `value_str` on that.

Reply:
And then we'd silently accept arbitrary other meta items, right? Currently the `rustc_attr!` can describe the `template!` of the attribute which AFAIK gets us some checks that reject any other shapes; but for custom inner meta item lists I don't think there is a corresponding `template!`?

I'm not convinced that is worth the effort, but if the reviewer insists I can do it.